bot/exts/moderation/silence.py

```diff
@@ -93,87 +93,229 @@ async def _async_init(self) -> None:
         await self.bot.wait_until_guild_available()
 
         guild = self.bot.get_guild(Guild.id)
-        self._verified_role = guild.get_role(Roles.verified)
+
+        self._verified_msg_role = guild.get_role(Roles.verified)
+        self._verified_voice_role = guild.get_role(Roles.voice_verified)
+        self._helper_role = guild.get_role(Roles.helpers)
+
         self._mod_alerts_channel = self.bot.get_channel(Channels.mod_alerts)
+
         self.notifier = SilenceNotifier(self.bot.get_channel(Channels.mod_log))
         await self._reschedule()
 
+    async def _get_related_text_channel(self, channel: VoiceChannel) -> Optional[TextChannel]:
+        """Returns the text channel related to a voice channel."""
+        # TODO: Figure out a dynamic way of doing this
+        channels = {
+            "off-topic": Channels.voice_chat,
+            "code/help 1": Channels.code_help_voice,
+            "code/help 2": Channels.code_help_voice_2,
+            "admin": Channels.admins_voice,
+            "staff": Channels.staff_voice
+        }
+        for name in channels.keys():
+            if name in channel.name.lower():
+                return self.bot.get_channel(channels[name])
+
+    async def send_message(
+        self, message: str, source_channel: TextChannel, target_channel: Union[TextChannel, VoiceChannel],
+        alert_target: bool = False
+    ) -> None:
+        """Helper function to send message confirmation to `source_channel`, and notification to `target_channel`."""
+        # Get TextChannel connected to VoiceChannel if channel is of type voice
+        voice_chat = None
+        target_is_voice_channel = isinstance(target_channel, VoiceChannel)
+
+        if target_is_voice_channel:
+            voice_chat = await self._get_related_text_channel(target_channel)
+
+        # Reply to invocation channel
+        source_reply = message
+        if source_channel != target_channel:
+            source_reply = source_reply.replace("current channel", target_channel.mention)
+        await source_channel.send(source_reply)
+
+        # Reply to target channel
+        if alert_target and source_channel not in [target_channel, voice_chat]:
+            if target_is_voice_channel:
+                if voice_chat is not None:
+                    await voice_chat.send(message.replace("current channel", target_channel.mention))
+
+            else:
+                await target_channel.send(message)
```

Comment:
This can be simplified

```suggestion
        # Reply to invocation channel
        source_reply = message
        if source_channel != target_channel:
            source_reply = source_reply.replace("current channel", target_channel.mention)
        await source_channel.send(source_reply)

        # Reply to target channel
        if alert_target:
            if isinstance(target_channel, VoiceChannel):
                voice_chat = await self._get_related_text_channel(target_channel)
                if voice_chat and source_channel != voice_chat:
                    await voice_chat.send(message.replace("current channel", target_channel.mention))
            elif source_channel != target_channel:
                await target_channel.send(message)
```